x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/add_layer.tsx

```diff
@@ -111,7 +120,26 @@ export function AddLayerButton({
               return {
                 toolTipContent,
                 disabled,
-                name: label,
+                name:
+                  type === layerTypes.ANNOTATIONS ? (
+                    <EuiFlexGroup gutterSize="xs">
+                      <EuiFlexItem grow={false}>
+                        <span className="lnsLayerAddButtonLabel">{label}</span>
+                      </EuiFlexItem>
+                      <EuiFlexItem>
+                        <EuiBadge color="hollow">
+                          <span className="lnsLayerAddButtonTechBadge">
+                            {i18n.translate('xpack.lens.configPanel.experimentalLabel', {
+                              defaultMessage: 'Technical preview',
+                            })}
+                          </span>
+                        </EuiBadge>
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
+                  ) : (
+                    <span className="lnsLayerAddButtonLabel">{label}</span>
+                  ),
+                className: 'lnsLayerAddButton',
```

Comment:
Suggest adding some additional width here to account for the addition of the badge and the previously suggested `EuiFlexGroup` gutter size increase, which gives the badge some breathing room.

```suggestion
                className: 'lnsLayerAddButton',
                width: 300,
```